tests/baselines/reference/initializerReferencingConstructorLocals.errors.txt

```diff
@@ -1,18 +1,16 @@
 tests/cases/conformance/classes/propertyMemberDeclarations/initializerReferencingConstructorLocals.ts(4,9): error TS2304: Cannot find name 'z'.
 tests/cases/conformance/classes/propertyMemberDeclarations/initializerReferencingConstructorLocals.ts(5,15): error TS2304: Cannot find name 'z'.
 tests/cases/conformance/classes/propertyMemberDeclarations/initializerReferencingConstructorLocals.ts(6,14): error TS2339: Property 'z' does not exist on type 'C'.
-tests/cases/conformance/classes/propertyMemberDeclarations/initializerReferencingConstructorLocals.ts(7,15): error TS1003: Identifier expected.
-tests/cases/conformance/classes/propertyMemberDeclarations/initializerReferencingConstructorLocals.ts(7,20): error TS2339: Property 'z' does not exist on type 'C'.
+tests/cases/conformance/classes/propertyMemberDeclarations/initializerReferencingConstructorLocals.ts(7,15): error TS2304: Cannot find name 'this'.

```

Comment:
That's because `parseEntityName` now allows reserved words anywhere, not just on the right of a dot. That affects even single identifiers, but in this case it now allows `this` in `this.x` to be parsed as an identifier.

Other callers that pass `allowReservedWords: true` are `parseTypeQuery` and `parseIsolatedEntityName`. Plus `parseTypeReference` in the case of JSDoc.

I think the change is fine in `parseTypeQuery` since this error is at least as easy to read as the old one. `parseIsolatedEntityName` just seems to be used for parsing `--jsxFactory`, so that change should be fine too (although I'm not sure why it allows reserved words at all).